client.go

```diff
@@ -554,6 +554,19 @@ func (c *Client) AddRetryCondition(condition RetryConditionFunc) *Client {
 	return c
 }
 
+// AddRetryAfterErrorCondition adds the basic condition of retrying after enncoutering
```

Comment:
@gtpaulose PR looks good, can you please correct typo `enncoutering` -> `encountering`?